autoload/sj/go.vim

```diff
@@ -52,9 +52,62 @@ function! sj#go#SplitStruct()
 endfunction
 
 function! sj#go#JoinStruct()
+  return s:joinStructOrFunc('{', '}')
+endfunction
+
+function! sj#go#SplitFunc()
+  let line = getline('.')
+  if line !~ '^func '
+    return 0
+  endif
+
+  let [start, end] = s:locateFuncBracesOnLine(line)
+  if start < 0 && end < 0
+    return 0
+  endif
+
+  let parsed = sj#ParseJsonObjectBody(start + 1, end - 1)
+
+  let arg_groups = []
+  let typed_arg_group = ''
+  for elem in parsed
+    if match(elem, '\s\+') != -1
+      let typed_arg_group .= elem
+      call add(arg_groups, typed_arg_group)
+      let typed_arg_group = ''
+    else
+      " not typed here, add to group
+      let typed_arg_group .= elem . ', '
+    endif
+  endfor
```

Comment:
It seems to me like the `typed_arg_group` variable is unused. Is it necessary to detect the types of the variables? It seems like we can split them easily without breaking them down into types and vars.